src/protections/WordList.ts

```diff
@@ -84,11 +84,19 @@ export class WordList extends Protection {
                     return
                 }
             }
-            if (this.badWords === null) {
-                // Create a mega-regex from all the tiny baby regexs
+            if (!this.badWords) {
+                const escapeRegExp = (string: string) => {
+                    return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+                };
+
+                // Create a mega-regex from all the tiny words.
+                const words = mjolnir.config.protections.wordlist.words.filter(word => word.length !== 0).map(escapeRegExp);
+                if (words.length === 0) {
+                    throw new Error("Someone turned the WordList protection on without configuring any words.");
+                }
                 try {
                     this.badWords = new RegExp(
-                        "(" + mjolnir.config.protections.wordlist.words.join(")|(") + ")",
+                        words.join("|"),
                         "i"
                     );
                 } catch (ex) {
```

Comment:
trying to work out what we expect `new RegExp` to throw

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/RegExp#exceptions it doesn't, but something makes me believe i was asked to catch this in an earlier review...

Reply:
Hmm https://github.com/matrix-org/mjolnir/pull/334#discussion_r919921553

Reply:
Well that was bad

Reply:
Not sure I understand this conversation. `new RegExp(...)` can throw, no?

Reply:
Well we don't support sub regexes and these are just words. So we shouldn't come across `pattern cannot be parsed as a valid regular expression.`, as well as `flags contains repeated characters or any character outside of those allowed.`

Reply:
Do we have something that validates that these are just words? Because, for instance, the following would raise an exception:

```js
new RegExp("[")
```

Of course, with your new code and escaping, that shouldn't happen.